### ISSUE TITLE
What is the best way to ensure that a composite ID is unique?

### ISSUE THREAD

**Issue by BrianGenisio** [graphql-python/graphene#716]
If we have a type which has a composite key, what is the best way to define it in Graphene?

Should serialize it ourselves?  Like:

```
   self.id = '{}/{}'.format(id_a, id_b)
```

Or should we let Graphene serialize an object for us?
```
   self.id = {
      'a': id_a,
      'b': id_b,
   }
```

Or should we do something different?

Is there a benefit to either approach?  We're inclined to go with our own custom key because we think it is most predictable... but we weren't sure if there is a convention which is preferred over custom serialization of composite keys.  

Thoughts?

**Comment by jkimbo** [comment]
@BrianChapman I'm not sure I fully understand the question but it's up to you to define whatever fields you might on a type even if the underlying data is a composite one. For example you might define your composite type like this:

```python
class CompositeType(graphene.ObjectType):
	id = graphene.ID(required=True)
	
	def resolve_id(root, info):
		return '{}:{}'.format(root['id_a'], root['id_b'])
```

If you want a globally unique ID (for example for using with Relay) you can use the `graphene.relay.Node` interface which will give you one.

Hope that helps!

**Comment by BrianGenisio** [comment]
@jkimbo Yes, this helps, and it seems you understand the questions fine.  I was just trying to understand if there were any conventions built in to the serialization of ID types that we should be adhering to, and it seems the answer is "no" which is a fine answer :)

Thanks!

**Comment by jkimbo** [comment]
@BrianGenisio did I manage to answer your question? Can this issue be closed if I did?
